website/docs/cdktf/debugging.mdx

```diff
@@ -0,0 +1,63 @@
+---
+page_title: Debugging
+description: >-
+  Turn on logging and troubleshoot problems in your CDKTF application.
+---
+
+# Debugging
+
+## Logging
+
+You can get more information about what CDKTF is doing and what problems have occured by setting the `CDKTF_LOG_LEVEL` environment variable to `debug`. If you want the logs written to a file you can set `CDKTF_LOG_FILE_DIRECTORY` to a file path, a `cdktf.log` file with all logs will be written. This can be helpful if you need to share them in e.g. a bug report.
+
+## Troubleshooting
+
+To understand where the problem is coming from it's a good first step to see if it's a synth-time issue or an execution-time issue.
+If `cdktf synth` fails you have a synth-time issue, otherwise it's an execution-time issue.
+
+## Synth-time issues
+
+The most important thing is to understand which construct is causing the problem. The error message thrown might hint you towards it, if not you will need to search for it. If you have a last version that successfully synthesized you can search through the new changes, otherwise you need to search through all changes. To search it you can repeatedly run `cdktf synth` and comment parts out until you find the error. The synthesized code does not need to make any logical sense since we don't plan to apply it.
+
+Once you find the construct causing the problem you need to try to isolate the problem. If you use variables in your constructs configuration, try to temporarily replace them with static values to ensure the problem is on the construct itself. If the problem persists try looking at the type to identfy missing or mistyped fields. Looking at the resource in the Terraform provider might help to understand the structure better as well.
+
+If this does not help, please create an issue in the [hashicorp/terraform-cdk](https://github.com/hashicorp/terraform-cdk/issues/new?assignees=&labels=bug%2C+new&template=bug-report.md&title=) repository.
+
+## Execution-time issues
+
+Execution-time issues can arise from misconfigured synthesized code or from the provider itself.
+As a first step you can check if the synthesized code looks like you would expect it. You can check through the Terraform documentation for the resource to see if there are misconfigured values. You can also search for the resource name, Terraform and the error to find issues Terraform users had with the resource or look at the providers Github repository for issues similar to yours. Keep in mind that in CDKTF renames resources and fields in most languages to be camel case, whereas Terraform uses snake case.
```

Comment:
> Are there any common ways that CDKTF can misconfigure the code?

There is no common way I'd say. CDKTF synthesizes what you tell it to synthesize, but you might have skipped reading through the provider documentation or you might have used the wrong name for your AMI or what not. There are unlimited possibilities to what could happen. It's a bit like basic programming debugging, does the output look like what I'd expect it to be.

> ill folks get error messages from Terraform itself along with CDKTF-specific errors?

They get the full terraform logs and if possible we annotate the cdktf construct path to the individual resources.

> is that something that can cause problems with the config? Or are you telling users that so they can know to ignore it when they're reviewing the code for errors?

in 99% of the cases this is a synth time problem since the compiler or JSII will tell you about it. It's not a source or errors, but more sth that could be confusing to users, like "I put 'clusterOptions' into my cdktf program but I can't find it in the JSON --> because it's 'cluster_options' in the JSON"